db/models/models.js

```diff
@@ -6,23 +6,29 @@ exports.locateTopics = (stuff) => {
   });
 };
 
-exports.locateArticleId = (article_Id) => {
-  if (article_Id > 500) {
+exports.locateArticleId = async (article_Id) => {
+  const queryArticles = db.query(
+    `SELECT * FROM articles WHERE article_id = ${article_Id};`
+  );
+  const result = await queryArticles;
+
+  if (result.rows.length === 0) {
     return Promise.reject({ status: 404, msg: "not found" });
   }
 
-  console.log(article_Id, "fail test");
-  let querys = `SELECT * FROM articles WHERE article_id = ${article_Id};`;
-  console.log(querys);
-  return db.query(querys).then((results) => {
-    return results.rows[0];
-  });
+  const queryComments = db.query(
+    `select * from comments where author = '${result.rows[0].author}';`
+  );
+  const newResult = await queryComments;
+
+  result.rows[0]["comment_count"] = newResult.rows.length;
+
+  return result.rows[0];
```

Comment:
Waiting for individual, separate database queries is going to be on the slower side, so whenever we can do all the requisite logic in one query, we should.

In this case, you should have a look at using the `COUNT` aggregate function with a `GROUP BY` to get the `comment_count` as part of the same query where you're getting the article information.